src/tlo/bitset_handler/bitset_extension.py

```diff
@@ -0,0 +1,730 @@
+    to be cast to their bytestring representation being being assigned.
+
+    Supported Operations (slices)
+    -----------------------------
+    When operating on slices or masks of the series, we have to re-implement the desired operators
+    so that users can continue to pass ``set``s as scalar arguments on the left. As a general rule
+    of thumb, if a binary operator can be performed on ``set``s, it will also work identically,
+    but entry-wise, on a bitset series.
+
+    ``NodeType`` instances will be cast to ``set``s if provided as singletons. Comparisons will be
+    performed entry-wise if a suitable vector of values is provided as the comparison target.
+
+    Currently implemented methods are:
+
+    = :
+        Directly assign the value on the right to the entry/entries on the left.
+    +, | :
+        Perform union of the values on the left with those on the right.
+    +=, |= :
+        In-place union; add values on the right to the sets on the left.
+    & :
+        Perform intersection of the values on the left with those on the right.
+    &= :
+        In-place intersection; retain only elements on the left that appear on the right.
+    -, -= :
+        Remove the values on the right from the sets on the left.
+    <, <= :
+        Entry-wise subset (strict subset) with the values on the right.
+    >, >= :
+        Entry-wise superset (strict superset) with the values on the right.
+        Note that the >= operation is the equivalent of entry-wise "if the values on the right
+        are contained in the bitsets on the left".
+    """
+
+    _data: NDArray[np.bytes_]
+    _dtype: BitsetDtype
+
+    @staticmethod
+    def uint8s_to_byte_string(arr: np.ndarray[np.uint8]) -> NDArray[np.bytes_]:
+        """
+        Returns a view of an array of ``np.uint8``s of shape ``(M, N)``
+        as an array of ``M`` fixed-width byte strings of size ``N``.
+        """
+        fixed_width = arr.shape[1]
+        return arr.view(f"{fixed_width}S").squeeze()
+
+    @classmethod
+    def _concat_same_type(cls, to_concat: Sequence[BitsetArray]) -> BitsetArray:
+        concat_data = np.concatenate(bsa._data for bsa in to_concat)
+        return cls(concat_data, to_concat[0].dtype)
+
+    @classmethod
+    def _from_sequence(
+        cls, scalars: Iterable[Set[NodeType]], *, dtype: BitsetDtype | None = None, copy: bool = False
+    ) -> BitsetArray:
+        """
+        Construct a new BitSetArray from a sequence of scalars.
+
+        Parameters
+        ----------
+        scalars : Sequence[Set[NodeType] | NodeType]
+            Sequence of sets of elements (or single-values to be interpreted as single-element sets)
+        dtype : dtype, optional
+            Cast to this datatype, only BitsetDtype is supported if not None. If None, an attempt will be made to
+            construct an appropriate BitsetDtype using the scalar values provided.
+        copy : bool, default False
+            If True, copy the underlying data.
```

Comment:
I think this is a NumPy style docstring parameter list - ideally we should use normal Sphinx / rST type with `:param name:` directives. Also think type hints should make type descriptions for parameters redundant